### PR TITLE
Create bitcoinabuse.py

### PR DESCRIPTION
**Pull request by threathive**
Analytics plugin which can lookup and tag known bad BTC addresses.

### CODE REVIEW

File: plugins/analytics/public/bitcoinabuse.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            'name': 'Bitcoinabuse.com API Key',
+            'description': 'API Key for bitcoinabuse.com.'
+        }
+    }
+
+    API = 'https://www.bitcoinabuse.com/api/'
+
+    @staticmethod
+    def get_abuse_type(abuse_type_id):
+    # This is for future use once they update their api.
```
Comment (tomchop) [inline plugins/analytics/public/bitcoinabuse.py]:
```suggestion
        # This is for future use once they update their api.
```

File: plugins/analytics/public/bitcoinabuse.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    # Currently they only return counts for reports.
```
Comment (tomchop) [inline plugins/analytics/public/bitcoinabuse.py]:
```suggestion
        # Currently they only return counts for reports.
```

File: plugins/analytics/public/bitcoinabuse.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    # If they start returning the abuse types we can get the mapping here.
```
Comment (tomchop) [inline plugins/analytics/public/bitcoinabuse.py]:
```suggestion
        # If they start returning the abuse types we can get the mapping here.
```

File: plugins/analytics/public/bitcoinabuse.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+        r = requests.get('{api_url}abuse-types'.format(api_url=BitCoinAbuseAPI.API) , proxies=yeti_config.proxy)
```
Comment (tomchop) [inline plugins/analytics/public/bitcoinabuse.py]:
```suggestion
        r = requests.get(
            '{api_url}abuse-types'.format(api_url=BitCoinAbuseAPI.API),
            proxies=yeti_config.proxy)
```

File: plugins/analytics/public/bitcoinabuse.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        if r.ok:
+            abuse_types = r.json()
+            for _type in abuse_types:
+                if _type.get('id') == abuse_type_id:
+                    return _type.get('label')
+
+    @staticmethod
+    def check(observable, api_key):
+        if isinstance(observable, Bitcoin):
```
Comment (tomchop) [inline plugins/analytics/public/bitcoinabuse.py]: Easy guard statement here:

```python
if not isinstance(observable, Bitcoin):
    return
url = '{api_url}reports...'
```

File: plugins/analytics/public/bitcoinabuse.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    ACTS_ON = ['Bitcoin']
+
+    @staticmethod
+    def analyze(observable, results):
+        links = set()
+        json_result = BitCoinAbuseAPI.check(
+            observable,
+            results.settings['bitcoinabuse_apikey']
+        )
+        if json_result:
```
Comment (tomchop) [inline plugins/analytics/public/bitcoinabuse.py]: Guard statement (although it's not necessary here)

**Comment by threathive** [comment]
I am gonna replaced this with an  InlineAnalytics plugin so I will close this and just submit another pr later.

**Pull request closed without merging**